src/components/semantic/presenters/LLMQuestionPresenter.tsx

```diff
@@ -66,11 +79,52 @@ export function LLMQuestionPresenter(props: PresenterProps<IsaacLLMFreeTextQuest
         });
     }
 
+    function evaluateMarkingFormula<T extends keyof LLMFreeTextMarkedExample>(markingFormula: LLMFormulaNode, value: LLMFreeTextMarkedExample[T]): number { 
+        if (isLLMConstantNode(markingFormula)) { 
+            return markingFormula.value; 
+        } else if (isLLMVariableNode(markingFormula)) {
+            if (typeof value === 'object') {
+                return value[markingFormula.name] ?? 0;
+            }
+            return 0;
+        } else if (isLLMFunctionNode(markingFormula)) {
+            const args: LLMFormulaNode[] = markingFormula.arguments;
+            switch (markingFormula.name) {
+                case "SUM":
+                    return args.map((arg: LLMFormulaNode) => evaluateMarkingFormula(arg, value)).reduce((acc: number, val: number) => acc + val, 0);
+                case "MAX":
+                    return Math.max(...args.map((arg: LLMFormulaNode) => evaluateMarkingFormula(arg, value)));
+                case "MIN":
+                    return Math.min(...args.map((arg: LLMFormulaNode) => evaluateMarkingFormula(arg, value)));
+                default:
+                    throw new Error("Unknown marking function: " + markingFormula.name);
+            }
+        }
+        throw new Error("Unknown marking expression type: " + markingFormula.type);
+    }
+
+    function evaluateMarkTotal<T extends keyof LLMFreeTextMarkedExample>(markingFormula: LLMFormulaNode, value: LLMFreeTextMarkedExample[T]): number {
+        if (markingFormula === undefined && typeof value === 'object' && value !== null) {
+            let total: number = 0;
+            for (const key in value) {
+                total = total + (value[key] ?? 0);
+            }
+
+            return Math.min(doc.maxMarks ?? 0, total);
```

Comment:
This one is intentional, yes - I've discussed it with Ada already. A question should always have maxMarks set in order to be valid since it's one of the pieces of data we send to the LLM, and if it _is_ uninitialised Java would define it as 0 in the equivalent equation in the API. This is the most consistent behaviour. 

It's already set for all published questions, so we don't really have any issue regardless :)